cql3/expr/expression.hh

```diff
@@ -717,6 +717,29 @@ bool is_single_column_restriction(const expression&);
 
 // Gets the only column from a single_column_restriction expression.
 const column_value& get_the_only_column(const expression&);
+
+// A comparator that orders columns by their position in the schema
+// For primary key columns the `id` field is used to determine their position.
+// Other columns are assumed to have position `123456789`.
+// In case the position is the same they are compared by their name.
+// This comparator has been used in the original restricitons code to keep
+// restrictions for each column sorted by their place in the schema.
+struct schema_pos_column_definition_comparator {
+    bool operator()(const column_definition* def1, const column_definition* def2) const;
+};
+
+// Extracts column_defs from the expression and sorts them using schema_pos_column_definition_comparator.
+std::vector<const column_definition*> get_sorted_column_defs(const expression&);
+
+// Extracts column_defs and returns the last one according to schema_pos_column_definition_comparator.
+const column_definition* get_last_column_def(const expression&);
+
+// A map of single column restrictions for each column
+using single_column_restrictions_map = std::map<const column_definition*, expression, schema_pos_column_definition_comparator>;
+
+// Extracts map of single column restrictions for each column from expression
+single_column_restrictions_map get_single_column_restrictions_map(const expression&);
+
```

Comment:
For later - we should reorganize it. expression.hh should contain the basic definitions and utilities, targeted functions should be moved out to narrower scope headers.